### DISCUSSION TITLE
Incremental Static Generation not persisting

### DISCUSSION THREAD

**Discussion by edwjusti** [vercel/next.js#17637] [Help]
# Bug report

## Describe the bug

Incremental Static Generation is triggered again after Next.js is restarted.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Clone this sample project https://github.com/lfades/static-tweet/blob/master/pages/index.js
2. Add a console.log with the tweet ID on this line: https://github.com/lfades/static-tweet/blob/3105e53bce14fa9a8dd004fb9cd66732069c29c9/pages/%5Btweet%5D.js#L29
3. Start the process for the first time with next build and next start
4. Trigger the static generation once
5. Restart the process without rebuilding
6. Make another request with the same tweet ID that triggered static generation previously

## Expected behavior

Static generation shouldn't be triggered again after restart if static assets are within the revalidate age.

## System information

- OS: macOS
- Version of Next.js: 9.5.3
- Version of Node.js: 14.9.0

**Comment by lfades** [comment]
Moved to a discussion because this is not an issue with Next.js.

@edwjusti In development Next.js always executes `getStaticProps` after a change, `revalidation` is for production (you can also test it with `next build && next start`)

**Reply by edwjusti** [reply]
This is happening in production too. Sorry for the poor choice of words, but you should be able to see this problem happening by placing a log message on that line linked above.

**Reply by lfades** [reply]
Static generation won't be triggered again on production unless there's revalidation, in the static tweet demo it will never trigger again for the same path because it doesn't have revalidation set. Your steps looked like it was a development issue but if that's not the case feel free to provide a reproduction we can take a look at, we have many tests to make sure it works.

Some other notes here:
- `getStaticProps` will be called again if you visit a path that wasn't visited before.
- the static tweet demo is some versions behind the current Next.js version so if there's an issue please update Next.js first 🙏
